python/ray/ml/train/impl/data_parallel_trainer.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from typing import Dict, Callable, Optional
+
+from ray.ml.trainer import Trainer
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.train.trainer import GenDataset
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.train import BackendConfig
+from ray.util.annotations import DeveloperAPI
+
+logger = logging.getLogger(__name__)
+
+
+@DeveloperAPI
+class DataParallelTrainer(Trainer):
+    """A Trainer for data parallel training.
+
+    This Trainer runs the provided function on multiple Ray Actors. The
+    provided datasets are automatically sharded to each Ray actor running
+    the training function.
```

Comment:
Can you add an example of what kind of train_func to provide? It's not clear what the sharding means. Does each func know its rank?

Reply:
Updated